rubin_sim/scheduler/modelObservatory/kinem_model.py

```diff
@@ -605,19 +610,35 @@ def slew_times(
         return slewTime
 
     def visit_time(self, observation):
-        # How long does it take to make an observation. Assume final read can be done during next slew.
+        # How long does it take to make an observation.
         visit_time = (
             observation["exptime"]
             + observation["nexp"] * self.shuttertime
             + max(observation["nexp"] - 1, 0) * self.readtime
         )
         return visit_time
 
+    def shutter_stall(self, observation):
+        """Time we need to stall after shutter closes to let things cool down"""
+        result = 0.0
+        delta_t = observation["exptime"] / observation["nexp"]
+        if delta_t < self.two_motion_time:
+            result = self.two_motion_time - delta_t
+        return result
+
     def observe(self, observation, mjd, rotTelPos=None, lax_dome=True):
         """observe a target, and return the slewtime and visit time for the action
 
         If slew is not allowed, returns np.nan and does not update state.
         """
+        if (observation["nexp"] > 2) & (
+            observation["exptime"] / observation["nexp"] < self.two_motion_time
+        ):
+            warnings.warn(
```

Comment:
I'd suggest adding a logger to the observatory model and using `self.log.warning` instead.. 

You could add something like this to the class, init method:

```
import logging

...
class Kinem_model(object):
    ...
    def __init__(
        self, location=None, park_alt=86.5, park_az=0.0, start_filter="r", mjd0=0, log=None
    ):

    self.log = log.getChild(type(self).__name__) if log is not None else logging.getLogger(type(self).__name__)
```

Reply:
Looks like a logger would then print the warning for every single exposure. I like the `warnings.warn` behavior of only printing once.